src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -0,0 +1,1753 @@
+   *
+   * <p>Subclasses can override this method to extract additional non-standard characteristics from
+   * the result set, and override the characteristics determined using the standard JDBC metadata
+   * columns and supplied as parameters.
+   *
+   * @param resultSet        the result set
+   * @param id               the column identifier
+   * @param jdbcType         the JDBC type of the column
+   * @param typeName         the name of the column's type
+   * @param classNameForType the name of the class used as instances of the value when {@link
+   *                         ResultSet#getObject(int)} is called
+   * @param nullability      the nullability of the column
+   * @param mutability       the mutability of the column
+   * @param precision        the precision of the column for numeric values, or the length for
+   *                         non-numeric values
+   * @param scale            the scale of the column for numeric values; ignored for other values
+   * @param signedNumbers    true if the column holds signed numeric values; null if not known
+   * @param displaySize      the preferred display size for the column values; null if not known
+   * @param autoIncremented  true if the column is auto-incremented; null if not known
+   * @param caseSensitive    true if the column values are case-sensitive; null if not known
+   * @param searchable       true if the column is searchable; null if no; null if not known known
+   * @param currency         true if the column is a currency value
+   * @param isPrimaryKey     true if the column is part of the primary key; null if not known known
+   * @return the column definition; never null
+   */
+  protected ColumnDefinition columnDefinition(
+      ResultSet resultSet,
+      ColumnId id,
+      int jdbcType,
+      String typeName,
+      String classNameForType,
+      Nullability nullability,
+      Mutability mutability,
+      int precision,
+      int scale,
+      Boolean signedNumbers,
+      Integer displaySize,
+      Boolean autoIncremented,
+      Boolean caseSensitive,
+      Boolean searchable,
+      Boolean currency,
+      Boolean isPrimaryKey
+  ) {
+    return new ColumnDefinition(id, jdbcType, typeName, classNameForType, nullability, mutability,
+                                precision, scale,
+                                signedNumbers != null ? signedNumbers.booleanValue() : false,
+                                displaySize != null ? displaySize.intValue() : 0,
+                                autoIncremented != null ? autoIncremented.booleanValue() : false,
+                                caseSensitive != null ? caseSensitive.booleanValue() : false,
+                                searchable != null ? searchable.booleanValue() : false,
+                                currency != null ? currency.booleanValue() : false,
+                                isPrimaryKey != null ? isPrimaryKey.booleanValue() : false
+    );
```

Comment:
This is another example of a method that can be easily overridden to customize the behavior, without having to override or copy the bulk of the logic for querying this information from the database.